### ISSUE TITLE
Not using the parse tree?

### ISSUE THREAD

**Issue by podlomar** [PhilippeSigaud/Pegged#213]
Hi, I would like to ask whether it is possible to construct a parse tree myself similarly as it is possible in flex/bison where each rule match just calls your predefined function and allows you to produce whatever structure you want. I am parsing a simple LISP-like expressions so the result of parsing is basically a bunch of linked lists which I would like to construct myself instead of traversing the parse tree and constructing the lists from it.

**Comment by veelo** [comment]
Are [sematic actions](https://github.com/PhilippeSigaud/Pegged/wiki/Semantic-Actions) what you are looking for? I don't think it would make much difference though, because a parse tree is (needs to be) constructed anyway.

**Comment by podlomar** [comment]
Yes, a know about the semantic actions and as you say, they do not make much difference. So there is no way to avoid the construction of the parse tree I suspect...
